firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/ExtensionFunctions.kt

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx.serialization
+
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.DocumentReference
+
+/**
+ * Overwrites the document referred to by this {@code DocumentReference}. If the document does not
+ * yet exist, it will be created. If a document already exists, it will be overwritten.
+ *
+ * @param data The data to write to the document (the data must be a @Serializable Kotlin object).
+ * @return A Task that will be resolved when the write finishes.
+ */
+inline fun <reified T> DocumentReference.setData(data: T): Task<Void> {
```

Comment:
Once we have the component based approach, are we still going to need this function?

Reply:
We do not need this function anymore once we have the component based approach.
But since I think I might need this code as a `testutil` for integration test to verify my Kotlin solution's result is the same as java solution.

Reply:
If this method is needed only for tests, can you move it `testutil` or into the test directly?

Reply:
In my following PR (for encoder's component registrar), I have moved this extension to the testutil.
https://github.com/firebase/firebase-android-sdk/pull/3849/files#diff-d4d3bb7507af7dd41bde6d153cb5649c49f1cdfb3892bde1b3b34df31c8d18bf

Reply:
> If this method is needed only for tests, can you move it `testutil` or into the test directly?

In my following PR (for encoder's component registrar), I have moved this extension to the testutil.
https://github.com/firebase/firebase-android-sdk/pull/3849/files#diff-d4d3bb7507af7dd41bde6d153cb5649c49f1cdfb3892bde1b3b34df31c8d18bf